### PR TITLE
[ui] Raise warning if the user tries to delete all groups from table

### PR DESCRIPTION
**Pull request by sakshikukreja14**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -30,6 +30,7 @@
 bin/crashreporter
 bin/El-MAVEN
 bin/test-libmaven
+bin/node_bin
```
Comment (saifulbkhan) [inline .gitignore:33]: Hah! Thank you :)

File: src/gui/mzroll/tabledockwidget.cpp [hunk 2]
```diff
@@ -1158,6 +1158,32 @@ void TableDockWidget::deleteGroup(PeakGroup *groupX) {
   updateCompoundWidget();
 }
 
+bool TableDockWidget::deleteAllgroupsWarning() 
+{
+    QMessageBox *warning = new QMessageBox(this);
+    bool selectedOption;
+
+    auto htmlText = QString("<p><b>Are you sure you want to permanently erase all the "
+                            "groups from table?</b></p>");
```
Comment (saifulbkhan) [inline src/gui/mzroll/tabledockwidget.cpp:1167]: "... from _this_ table?"

File: src/gui/mzroll/tabledockwidget.cpp [hunk 3]
```diff
@@ -1158,6 +1158,32 @@ void TableDockWidget::deleteGroup(PeakGroup *groupX) {
...
+    htmlText += "<p>You can't undo this action.</p>";
```
Comment (saifulbkhan) [inline src/gui/mzroll/tabledockwidget.cpp:1168]: "You _cannot_ undo ..."

File: src/gui/mzroll/tabledockwidget.cpp [hunk 4]
```diff
@@ -1158,6 +1158,32 @@ void TableDockWidget::deleteGroup(PeakGroup *groupX) {
...
+    auto yesButton = warning->addButton(tr("Yes"),
+                                  QMessageBox::AcceptRole);
+    warning->exec();
+
+    if(warning->clickedButton() == yesButton)
+        selectedOption = true;
+    else
+        selectedOption = false;
+
+    QCoreApplication::processEvents();
```
Comment (saifulbkhan) [inline src/gui/mzroll/tabledockwidget.cpp:1183]: Why was this needed? If not for this, there would be no need for the `selectedOption` boolean and lines 1178 to 1184 can be reduced to:
```C++
if (warning->clickedButton() == yesButton)
    return true;

return false;
```

File: src/gui/mzroll/tabledockwidget.cpp [hunk 5]
```diff
@@ -1184,6 +1212,14 @@ void TableDockWidget::deleteSelectedItems()
...
         return;
 
+    // checks if the selected item count is same as the no. of 
+    // groups in the table.
+    if (topLevelItemsCount == topLevelGroupCount()) {
+        auto continueDeletion = deleteAllgroupsWarning();
+        if (!continueDeletion) {
+          return;
+        }
+    }
```
Comment (saifulbkhan) [inline src/gui/mzroll/tabledockwidget.cpp:1222]: Issue #853 was about warning the user when the select "Delete All Groups" from the context menu as well. This check can still be useful. If the `topLevelItemsCount == topLevelGroupCount()` is true then you simply call `TableDockWidget::deleteAll` method, instead of the complex logic that follows which is totally unnecessary since the user has asked to indiscriminately wipe out each item the table.

**Review by saifulbkhan — APPROVED** [review]
Good work. Please rebase.

**Comment by sakshikukreja14** [comment]
Done :)
